src/common/dependency-manager/local-manager.ts

```diff
@@ -178,4 +178,39 @@ export default class LocalDependencyManager extends DependencyManager {
     }
     super.loadParameters();
   }
+
+  /**
+   * @override
+   */
+  async loadExternalService(env_service_config: EnvironmentService, service_ref: string) {
+    if ('interfaces' in env_service_config) {
+      for (const [name, interface_details] of Object.entries(env_service_config.interfaces || {})) {
+        if (!interface_details.host || !interface_details.port) {
+          throw new Error(`As an interface specified in the environment config, interface ${name} requires that both a host and port be declared.`);
+        }
+      }
+
+      if (env_service_config.debug?.path) {
```

Comment:
Hmm not sure what to do about this - prob not worth overriding to only validate local. Would prefer this method to be the same for local/remote.

I wonder if we need to have a method on manager to load a config (local or remote). Also was hoping to have the graph validate itself so maybe we need to store service_config on ExternalNode. Thoughts?

Reply:
I went back on this one a few times. Really the only reason that this function is overridden here is because the external node has no idea what it is and needs to find out one way or another. I like the idea of having some kind of whatIsMyServiceConfig function. By doing so, we would also be able to store a service config on an external node like you mentioned. It could also potentially be useful in other places? I'll test it out tomorrow and see what it would look like

Reply:
lets add a method to the dependency manager
`loadServiceConfig(node_ref:string)` (loads from api by default)

then override in local dependency manager
```
loadServiceConfig(node_ref:string) {
  if (this.environment.services[node_ref]?.debug?.path) {
    // load from local path
  } else {
    return super.loadServiceConfig(node_ref);
  }
}
```

Reply:
then we can use this method in the base dependency_manager loadExternalService

Reply:
use this method in the loadService and loadLocalService methods too

Reply:
One thing about the base dependency manager is that it doesn't have an instance of an api. Should the method be abstract in the base manager? Then it will either load from the path or the api in the local manager, or the database in the server manager?

Reply:
Okay keep it as an abstract method and define on the local